plonky2x/src/frontend/builder/mux.rs

```diff
@@ -9,7 +9,25 @@ trait MuxBuilder {
 
 impl<F: RichField + Extendable<D>, const D: usize> MuxBuilder for CircuitBuilder<F, D> {
     fn select_index<T: CircuitVariable>(&mut self, selector: Variable, inputs: &[T]) -> T {
-        todo!()
+        let num_var_in_t = inputs[0].variables().len();
```

Comment:
This is great to have this implementation in addition to your other one. Do you want to add this implementation to the other PR as another method `select_index_with_gate` and then benchmark/compare the 2 methods? I'm personally curious to see which one will come out ahead.